### PR TITLE
[API] Now validate using create_timepoint->validate 

### PR DESCRIPTION
**Pull request by xlecours**
This pull request use the valide method of create_timepoint class to validate inputs befor timepoint creation. It calls the _valide() method of the create_timepoint module class.

### CODE REVIEW

File: htdocs/api/v0.0.2/candidates/Visit.php [hunk 1]
```diff
@@ -15,10 +15,12 @@
     get_include_path()
     . ":" . __DIR__ . "/../"
     . ':' . __DIR__ . "/../../../../php/libraries/"
+    . ':' . __DIR__ . "/../../../../modules/create_timepoint/php/"
```
Comment (driusan) [inline htdocs/api/v0.0.2/candidates/Visit.php]: Requires into a different module adds a bit of a code and has been fickle when we've done it before.

Wouldn't it make more sense to move the validation logic to the TimePoint class and call it from both places?

Comment (xlecours) [inline htdocs/api/v0.0.2/candidates/Visit.php]: Yes I concur.

**Comment by kongtiaowang** [comment]
PHPCS

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 2]
```diff
@@ -218,51 +219,25 @@ class Create_Timepoint extends \NDB_Form
...
-
-        //If the visitLabel is already in use then let the user pick another
-        foreach ($timePointArray AS $used_label) {
-            if (strcasecmp($values['visitLabel'], $used_label) == 0) {
-                $errors['visitLabel'] = 'This visit label is not unique.';
-            }
+        try {
+            \TimePoint::isValidVisitLabel($candid, $subprojectID, $visitLabel);
+        } catch ( \LorisException $e) {
+            $errors[] = $e->getMessage();
```
Comment (driusan) [inline modules/create_timepoint/php/create_timepoint.class.inc]: Shouldn't this be `$errors['visitLabel']`? (That's where the error was put in the old code being deleted..)

**Comment by xlecours** [comment]
@driusan I could change the function name for validateVisitLabel instead of is... The previous implementation is expecting an array of errors upon failure so the binary output is already compromised

**Pull request merged**